packages/compass-aggregations/src/components/add-stage/add-stage.tsx

```diff
@@ -6,41 +6,89 @@ import {
   css,
   spacing,
   Link,
+  palette,
 } from '@mongodb-js/compass-components';
 import { PIPELINE_HELP_URI } from '../../constants';
+import { useDroppable } from '@dnd-kit/core';
+
+const useCaseDropMarkerStyles = css({
+  width: '100%',
+  height: spacing[1],
+  borderRadius: spacing[1],
+  background: palette.green.dark1,
+  position: 'absolute',
+});
+
+type UseCaseDropMarkerProps = {
+  id: number;
+  style?: React.CSSProperties;
+};
+
+const UseCaseDropMarker = (props: UseCaseDropMarkerProps) => {
+  const { setNodeRef, isOver, active } = useDroppable({ id: props.id });
+  const draggedElementIsUseCase = active?.data.current?.type === 'use-case';
+  const draggedUseCaseIsOverMarker = isOver && draggedElementIsUseCase;
+  const markerInlineStyles: React.CSSProperties = {
+    ...props.style,
+    visibility: draggedUseCaseIsOverMarker ? 'visible' : 'hidden',
+  };
+
+  return (
+    <div
+      ref={setNodeRef}
+      className={useCaseDropMarkerStyles}
+      style={markerInlineStyles}
+      data-testid={`use-case-drop-marker-${props.id}`}
+    />
+  );
+};
 
 const iconContainerStyles = css({
   textAlign: 'center',
   marginTop: spacing[1] / 2,
   marginBottom: spacing[1] / 2,
+  position: 'relative',
 });
 
 const buttonContainerStyles = css({
   textAlign: 'center',
   marginTop: spacing[4],
   marginBottom: spacing[3],
+  position: 'relative',
 });
 
 const linkContainerStyles = css({
   textAlign: 'center',
   marginTop: spacing[2],
   marginBottom: spacing[2],
+  position: 'relative',
 });
 
 type AddStageProps = {
+  index: number;
+  renderUseCaseDropMarker: boolean;
   variant: 'button' | 'icon';
   onAddStage: () => void;
 };
 
-export const AddStage = ({ onAddStage, variant }: AddStageProps) => {
```

Comment:
This component is not an AddStage anymore. Should we consider renaming it? Or add another component like `StageSeperator` or something, which can handle the logic of showing droppable area or `AddStage` button?

Reply:
That sounds good actually. I will take care of that. Thank you :)